observe/datadog.go

```diff
@@ -0,0 +1,96 @@
+package observe
+
+import (
+	"os"
+
+	datadog "github.com/DataDog/opencensus-go-exporter-datadog"
+	"github.com/pkg/errors"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/trace"
+)
+
+// IsDatadogEnabled checks if exporitng metrics and traces to Datadog should enabled (by
+// setting DATADOG_ENABLED environemnt variable) and if the Datadog's agent address is
+// provided (by DATADOG_ADDR environsmnt variable)
+func IsDatadogEnabled() bool {
+	return os.Getenv("DATADOG_ENABLED") != "" && getDatadogAddr() != ""
+}
+
+// RegisterAndObserveDatadog will initiate and register Datadog metrics and tracing in environments
+// that pass the tests in IsDatadogEnabled function
+func RegisterAndObserveDatadog(onError func(error)) error {
+	if SkipObserve() {
+		return nil
+	}
+
+	exp, err := NewDatadogExporter(onError)
+	if err != nil {
+		return errors.Wrap(err, "unable to initiate Datadog tracing exporter")
+	}
+	trace.RegisterExporter(exp)
+	view.RegisterExporter(exp)
+
+	return nil
+}
+
+// NewDatadogExporter will return the tracing and metrics through
+// the stack driver exporter, if exists in the underlying platform.
+// If exporter is registered, it returns the exporter so you can register
+// it and ensure to call Flush on termination.
+func NewDatadogExporter(onErr func(error)) (*datadog.Exporter, error) {
+	_, service, version := GetServiceInfo()
+
+	opts := getDatadogOpts(service, version, getDatadogAddr(), onErr)
+	if opts == nil {
+		return nil, nil
+	}
+
+	return datadog.NewExporter(*opts)
+}
+
+func getDatadogAddr() string {
+	return os.Getenv("DATADOG_ADDR")
+}
+
+// getDatadogOpts returns Datadog Options that you can pass directly
+// to the OpenCensus exporter or other libraries.
+func getDatadogOpts(service, version, datadogAddress string, onErr func(err error)) *datadog.Options {
+
+	canExport := IsDatadogEnabled()
+	if !canExport {
+		return nil
+	}
+
+	return &datadog.Options{
+		// Namespace specifies the namespaces to which metric keys are appended.
+		// TODO: Figure out what the namespace should be. Can be either a projectID or something else.
```

Comment:
is the namespace here used for metrics or for traces?

Reply:
Metrics https://github.com/DataDog/opencensus-go-exporter-datadog/blob/master/datadog.go#L53